add option: scriptAttribute

prasannavl opened the pull request:
`asyncScriptTags` limits only setting the scripts as `async`. There are cases where it's more beneficial to protect the ordering with `defer` instead. 

This option deprecated the `asyncScriptTags`, and instead provides a new option called `scriptAttribute` that takes any attribute, and set it to true. Obviously, the currently supported ones can be `async|defer|none`

stereobooster commented:
HI. After some [experiments with `asyncScriptTags`](https://github.com/stereobooster/react-snap/blob/master/doc/an-almost-static-stack-optimization.md#round-7-load-js-in-a-non-blocking-manner) I consider it useless. I would like rather deprecate some features

prasannavl commented:
I looked at the `fixWebpackChunksIssue` function. However it just seems to use a `preload`. While `preload` would be the ideal solution for the future. Unfortunately, at it's current state not all major browsers handle it correctly. So, I wonder if that really solves anything at this point.

stereobooster commented:
If `preload` doesn't work - browser will fallback to Webpack's JSONP loading mechanism. 

> Unfortunately, at it's current state not all major browsers handle it correctly.

[As far as I can see](https://caniuse.com/#search=preload) support for preload is almost here. Or I miss something?

prasannavl commented:
Firefox has just gotten it, and Edge is yet to for General Public Release. Jsonp loading mechanism is what makes it  Anyway, I suppose that should be find if you don't intend to support the older cases. Okay, closing.

stereobooster commented:
@prasannavl hey I mentioned this experiment with asyncScriptTags and how it is useless. Today I discovered that it is useless for Chrome (Android), but not for Safari (iOS). Mobile safari actually waits for script to download, before it starts paint. I need to update my experiment

prasannavl commented:
@stereobooster - Yup. In my custom build initially, I had it `defer` instead and removed async entirely. But I didn't still like that it felt a little hacky straying off upstream. So, introduced this https://github.com/stereobooster/react-snap/pull/170 PR, which I think handles all this quite neatly, and allows hacks specific to custom projects to remain separate, while common ones can make in here eventually.
